### PR TITLE
MP-1215 Add images without conda/pip dependencies

### PR DESCRIPTION
**Pull request by TimurDaukaevMoonvision**
## Description

This PR contains changes for the base images, which were needed for migrating `vid-vision` to `uv`:
* Since we have upgraded `pytorch`, we don't have to install an older `gcc` anymore, so I could switch base Debian images from Bullseye to Bookworm (11 -> 12).
* `opensc` had to be upgraded from `0.20.0` to `0.23.0` to work in newer Debian.
* I don't replace old conda-based images, but add new ones (we may still need old images somewhere).
* We always use a prebuilt `pytorch` now, even for cpu images.
* `genicam` images have been removed.
* For `uv` there are no separate `pytorch` images now: they are merged with `moonbox` images, since we don't do any python installations now.
* All python dependencies have been moved to corresponding repositories; base images do not contain them anymore.

## Notes
* I didn't try to optimize installing of `apt` packages except some cases, when I had to do changes for adding `uv` anyway.
* I could not completely get rid of conda: there were some packages, which otherwise have some complicated installation procedure - and I don't know, what exactly is needed from them (i.e., which parts should be installed, with which parameters should they be built etc.).

## How to test
Builds are successful. Actual testing will happen in PRs for `vid-vision` and `vid-api`

### CODE REVIEW

File: docker/python-base/DockerfileUV [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ADD https://astral.sh/uv/install.sh /uv-installer.sh
+RUN sh /uv-installer.sh && rm /uv-installer.sh
+ENV PATH="/root/.local/bin/:$PATH"
+ARG python_version
+RUN uv python install $python_version
+# Without this you will have to pass `--no-sync` for each `uv run` call, because it will try to install dev requirements otherwise
+ENV UV_NO_SYNC=true
+# Next two are needed, because otherwise uv cannot find certificates
+ENV UV_NATIVE_TLS=true
+ENV SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
```
Comment (ahirner) [inline docker/python-base/DockerfileUV:60]: Is there something similar to `micromamba clean -yaf`, as in cleaning uv cache after it installing packages?

File: docker/python-base/DockerfileUV [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  && apt-get purge --autoremove --yes python python3 \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+# Install uv
+ADD https://astral.sh/uv/install.sh /uv-installer.sh
+RUN sh /uv-installer.sh && rm /uv-installer.sh
+ENV PATH="/root/.local/bin/:$PATH"
+ARG python_version
+RUN uv python install $python_version
```
Comment (ahirner) [inline docker/python-base/DockerfileUV]: nit: does this leave some ´python` cached, and if so can it be cleaned in the same command? (similar to `micromamba clean -yaf`, except we don't install any actual requirements via uv AFAIU).

Comment (TimurDaukaevMoonvision) [reply docker/python-base/DockerfileUV]: It has `--no-cache` flag as well as `uv cache clean` command.

So, answering the question - I think I could simply use `--no-cache` for python installation, as I do it for `uv sync` in downstream images (and then `uv cache clean` hopefully will not be needed at all).

Comment (ahirner) [reply docker/python-base/DockerfileUV]: `--no-cache` sounds good

**Review by ahirner — APPROVED** [review]

**Pull request merged**